### ISSUE TITLE
Calendar Widget

### ISSUE THREAD

**Issue by saumier** [culturecreates/minimal-footlight-search-widget#81]
We would like to use the same widget for another 3rd party site. In this example, another web site in the region of Outaouais is called "MRC Collines-de-outaouais".

Features to add to html for installing the widget on another website:
- config to hide the navbar for event/workshop/organization
- config to keep event list open rather than opening/closing like on toutculture.ca
- links to "view all" on toutculture.ca should have the same number of total hits in both places
- config to set a filter by region ie. Tout culture wants to show only events and workshops in the region "collines-de-outouais".
- config to open new tab/window when clicking events or any click leaving the widget.  The Toutculture.ca site should open full screen (not inside the 3rd party site).

Additional request from Minimal for the search widget:
1. session cookie storage of the current search term, tab (event, workshop, org) and date or date range so that a new page load does not erase the search settings in progress.
2. Adjust the thumbail image to use the image's natural aspect ratio (remove fitimage: cover) but still set the max height.

![Image](https://github.com/culturecreates/minimal-footlight-search-widget/assets/419491/b78f4451-b3b7-4b5f-a474-570b7d888c40)

**Comment by troughc** [comment]
@SyamBabu-M please assign this issue to me after you do the estimate.

**Comment by SyamBabu-M** [comment]
@troughc I have some doubts regarding this tickets. Can we discuss this in todays standup?

**Comment by SyamBabu-M** [comment]
@saumier https://docs.google.com/spreadsheets/d/1NgK_C-MSRr4qEwTDhSCDiOtU6bfrq_eyKe7ZjKRcw7s/edit#gid=0

**Comment by saumier** [comment]
Estimate of 24 hours by Syam.

**Comment by saumier** [comment]
@SyamBabu-M I have updated the Readme with my suggested parameters and created a PR here https://github.com/culturecreates/minimal-footlight-search-widget/pull/82

Please review and let me know if you have any questions or ideas to do the configurations differently. You can merge the PR if you agree with the changes.

**Comment by SyamBabu-M** [comment]
@saumier I have merged the PR. The suggestions looked great to me.

**Comment by SyamBabu-M** [comment]
@troughc Is the design for this completed?

**Comment by SyamBabu-M** [comment]
@saumier @troughc 
http://demo.staging-tout-culture.s3-website.ca-central-1.amazonaws.com/
http://staging.minimal-footlight-search-widget.s3.ca-central-1.amazonaws.com/v2/test-dist.html

please use these links to test the widget.

**Comment by saumier** [comment]
@troughc The link with the 3 tabs is relevant because it is the new widget that will be seen on toutculture.ca.  So we need to test that the new widget works in both places at the same time.

I will update the parameters on the test page to show FR and the  region "collines-de-outouais" but it will take me a bit of time. I'll ping you when it is done.

**Comment by saumier** [comment]
The region ID for collines de l'outouais is 63bc0b2d1c6b6c005aad5253

**Comment by saumier** [comment]
@SyamBabu-M I did some testing and there are things that need to be fixed.  Below is my list divided into MUST FIX and NICE TO HAVE.  Please only do the must fix and after you have tested the must fix and you still have time then you can work on the nice to have.

MUST FIX
1. Unable to configure the filter by region. The widget must be configurable to only show events from the region "collines de l'outaouais".  This should be configurable with the parameter data-search-events-filter='&region=63bc0b2d1c6b6c005aad5253&exclude-type=64776b93fbeda20064d2332f'. There are today only 3 events and no workshops. Test by comparing with this result: https://toutculture.ca/filter-evenements/?region=63bc0b2d1c6b6c005aad5253
2. concept ids for event filters must no be hardcoded. You currently hardcode "type=64776b93fbeda20064d2332f" and "exclude-type=64776b93fbeda20064d2332f". These should be passed in to the widget using parameters like data-search-events-filter. You will probably need to add a new parameter for workshops like data-search-workshop-filter.
3. there is a delay when clicking the search box. It appears that the throbber starts and only after about 1 second then the Footlight API is called. The API should be called right away when you click on the search box, and only while the person is typing should there be a debounce.


NICE TO HAVE
1. The parameters inline and float are backwards. The float is when the search results are in a layer above the webpage, inline is when the search box and results all appear in the page together. The default is "float" so it remains compatible with toutculture.ca.
2. There appears to be only 9 events listed, but the API call has a limit of 10. Why are we only displaying 9 of the 10? Either the limit is decreased or the display increased.

**Comment by saumier** [comment]
The footer label says no events are found, but there are events.  Label is incorrect.

![Image](https://github.com/culturecreates/minimal-footlight-search-widget/assets/419491/e7231033-4a90-4838-a83c-755476c2847c)

**Comment by saumier** [comment]
@troughc I tested and it looks good. I am pushing to production.

**Comment by saumier** [comment]
@SyamBabu-M Please add links that you found here, and your thoughts on pros and cons.  Also, please try to setup a test page with the iframe passing the parameters.

**Comment by SyamBabu-M** [comment]
@saumier you can access the test page using the link given below.
http://staging.minimal-footlight-search-widget.s3.ca-central-1.amazonaws.com/v3/test-iframe.html

**Comment by troughc** [comment]
test according ot the MRC story:
- when you click results, open in a new tab
- inline 
- uses iFrame
- limiting set of filters (targets third party website)
- No organizations

**Comment by saumier** [comment]
@SyamBabu-M What is the status?  I need to let the client know when they can start using v3 on the Minimal site.  Does the default click in production now open in the same window?

**Comment by SyamBabu-M** [comment]
@saumier I have added a [PR](https://github.com/culturecreates/minimal-footlight-search-widget/pull/88)

**Comment by saumier** [comment]
@troughc Have you tried the widget on MRC with the click opening in the iframe instead of opening a new window? It is interesting. Maybe even better? Try it out and let me know what you think.

**Comment by troughc** [comment]
@saumier in my marketing experience  customers (i.e. a customer like Tout Culture) often use a widget like this as a way of:
- promoting events
- promoting their own calendar
So having the calendar open on another tab helps bring traffic to the Tout Cutlure website.
But, I can also talk with Isabelle at Tout Culture about this to get her feedback directly. Because it is interesting.
